plugin-flex-ts-template-v2/src/feature-library/send-audio-rec-file/custom-components/AudioRecorderPanel/AudioRecorderPanel.tsx

```diff
@@ -0,0 +1,166 @@
+        });
+
+        this.setState({ isRecording: false, audioFile: null });
+      } else {
+        console.log('No audio file to detach.');
+      }
+    } catch (error) {
+      console.log('Error during file detachment:', error);
+    }
+  };
+
+  sendRec = async (buffer: any, conversationSid: any) => {
+    const audioFile = new File(buffer, 'voice-recording.mp3', {
+      type: 'audio/mpeg',
+      lastModified: Date.now(),
+    });
+
+    Actions.invokeAction('AttachFiles', {
+      files: [audioFile],
+      conversationSid,
+    });
+
+    this.setState({ audioFile }, () => {
+      const updatedAudioFile = this.state.audioFile;
+      console.log('Updated audioFile:', updatedAudioFile);
+    });
+  };
+
+  componentDidMount() {
+    if (navigator.mediaDevices && navigator.mediaDevices.getUserMedia) {
+      navigator.mediaDevices
+        .getUserMedia({ audio: true })
+        .then(() => {
+          console.log('Permission granted');
+          this.setState({ isBlocked: false });
+        })
+        .catch(() => {
+          console.log('Permission denied');
+          this.setState({ isBlocked: true });
+        });
+    }
+  }
+
+  render() {
+    const { showRecorder } = this.props;
+    if (showRecorder) {
+      return (
+        <div>
+          <AudioRecorderWrapper>
+            <AudioRecorderPopover>
+              <Grid gutter="space40" vertical>
+                <Card padding="space40">
+                  <Text as="p" marginBottom="space30" color="colorTextWeak">
+                    {templates[StringTemplates.AudioRecorder]()}
+                  </Text>
+                  <Text as="p" marginBottom="space20"></Text>
+                  <audio src={this.state.blobURL} controls={!this.state.isRecording} />
+                  <Paragraph marginBottom="space0">
+                    {this.state.isRecording ? (
+                      <>
+                        <audio src={this.state.audioURL} controls={this.state.isRecording} />
+                        <div style={{ display: 'flex', alignItems: 'center' }}>
+                          <Text as="p" marginBottom="space20" style={{ marginRight: '10px' }}>
+                            {templates[StringTemplates.RecordingMessage]()}
+                          </Text>
+                          <Spinner decorative={false} title="Loading" color="colorTextError" />
+                        </div>
+                        <IconButton icon="MuteLargeBold" onClick={this.stopRec} />
+                      </>
+                    ) : (
+                      <React.Fragment>
+                        <IconButton icon="MuteLarge" onClick={this.startRec} />
+                        {this.state.audioFile ? (
+                          <>
+                            <IconButton icon="Close" onClick={this.dontSendRec} />
+                          </>
+                        ) : null}
+                      </React.Fragment>
+                    )}
+                  </Paragraph>
```

Comment:
I made some adjustments to simplify the layout (goes in concert with the `width` CSS change). What do you think?

```jsx
<Text as="p" marginBottom="space30" color="colorTextWeak">
  {templates[StringTemplates.AudioRecorder]()}
</Text>
<audio src={this.state.blobURL} controls={this.state.audioFile ?? false} />
<div style={{ display: 'flex', alignItems: 'center' }}>
  {this.state.isRecording ? (
    <>
      <IconButton icon="MuteLargeBold" onClick={this.stopRec} />
      <Text as="p" marginLeft="space30" marginRight="space30">
        {templates[StringTemplates.RecordingMessage]()}
      </Text>
      <Spinner decorative={true} color="colorTextError" />
    </>
  ) : (
    <>
      <IconButton icon="MuteLarge" onClick={this.startRec} />
      {this.state.audioFile ? (
        <IconButton
          icon="Close"
          onClick={this.dontSendRec}
          title={templates.LinkPanelRemoveButtonLabel()}
        />
      ) : null}
    </>
  )}
</div>
```

Reply:
I also think adding `title` for the record and stop recording buttons would be good.

Reply:
cool... I have added this and template strings for tittle for the buttons. 
Done